functions/Get-DbaClusterNode.ps1

```diff
@@ -0,0 +1,116 @@
+            This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+            Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+        .NOTES
+            Tags:
+            dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+            Copyright (C) 2016 Chrissy LeMaire
+            License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+        .LINK
+            https://dbatools.io/Get-DbaClusterNode
+
+        .EXAMPLE
+            Get-DbaClusterNode -SqlInstance sqlcluster
+
+            Returns all nodes in the cluster and details about each node.
+
+        .EXAMPLE
+            Get-DbaClusterNode -SqlInstance sqlcluster -ActiveNode
+
+            Returns the name of the active node in the cluster
+
+    #>
+    param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [switch]$ActiveNode,
+        [switch]$Detailed,
+        [alias('Silent')]
+        [switch]$EnableException
+    )
+
+    begin {
+        Test-DbaDeprecation -DeprecatedOn 1.0.0 -Parameter Detailed
+        $server = Connect-SqlInstance -SqlInstance $SqlInstance -SqlCredential $SqlCredential -RegularUser
+        $computername = $server.ComputerNamePhysicalNetBIOS
+    }
+
+    process {
+        if ($server.IsClustered -eq $false) {
+            Stop-Function -Message "Not a clusterd instance." -Category ConnectionError -ErrorRecord $_ -Target $SqlInstance -Continue
+        }
+
+        if ($ActiveNode) {
+            try{
+                if($server.VersionMajor -ge 11) {
+                    $sql = "SELECT * FROM sys.dm_os_cluster_nodes where is_current_owner = 1"
+                    $datatable = $server.ConnectionContext.ExecuteWithResults($sql).Tables
+                }
+                        [PSCustomObject]@{
+                            ComputerName      = if($server.VersionMajor -ge 11){$datatable.nodename} else {$computername}
+                            InstanceName      = $server.ServiceName
+                            SqlInstance       = $server.DomainInstanceName
+                            Status            = $datatable.Status
+                            StatusDescription = $datatable.StatusDescription
+                            CurrentOwner      = $datatable.is_current_owner
+                        } | Select-DefaultView -Property ComputerName
+            }
+            catch{
+                Stop-Function -Message "Unable to query sys.dm_os_cluster_nodes on $server." -ErrorRecord $_ -Target $SqlInstance -Continue
+            }
+        }
+        else {
+            try{
+                $sql = "SELECT * FROM sys.dm_os_cluster_nodes"
```

Comment:
I don't see the reason behind calling this if the version is less than 11? This looks redundant to me as-is.